src/components/modebar/manage.js

```diff
@@ -99,10 +99,19 @@ function getButtonGroups(fullLayout, buttonsToRemove, buttonsToAdd) {
 
     var hasCartesian = fullLayout._hasCartesian,
         hasGL2D = fullLayout._hasGL2D,
-        allAxesFixed = areAllAxesFixed(fullLayout);
+        allAxesFixed = areAllAxesFixed(fullLayout),
+        dragModeGroup = [];
+
+    if((hasCartesian || hasGL2D) && !allAxesFixed) {
+        dragModeGroup = ['zoom2d', 'pan2d'];
+    }
+    if(hasCartesian) {
+        dragModeGroup.push('select2d');
+        dragModeGroup.push('lasso2d');
```

Comment:
> I guess it depends how fast you think we'll add other trace types.

It doesn't look like it will be hard to add `selectPoints` for the other trace types, but I doubt that it will be a priority for the next few months.

I'd vote for a check a fullData.